### PR TITLE
feat: aleo application verifier

### PR DESCRIPTION
**Pull request by yjamin**
### Description
This PR adds the application verifier for aleo.

### Related issues
[Issues](https://linear.app/hyperlane-xyz/issue/ENG-2329/agent-application-verifier)

## Summary by CodeRabbit

* **New Features**
  * Added an Aleo runtime verifier that enforces message structure, token parsing, and requires a specific route marker.
* **Bug Fixes**
  * Clearer rejection of malformed Aleo messages (length, alignment, parse failures, and missing context).
* **Tests**
  * Unit tests covering body-size edge cases, context scenarios, and token parsing outcomes.

### CODE REVIEW

File: rust/main/chains/hyperlane-aleo/src/application.rs
[the remark this answers is not in the document]
Comment (yjamin) [reply rust/main/chains/hyperlane-aleo/src/application.rs]: Sure.

File: rust/main/chains/hyperlane-aleo/src/application.rs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+use derive_new::new;
+use tracing::trace;
+
+use hyperlane_core::{Decode, HyperlaneMessage};
+use hyperlane_operation_verifier::{
+    ApplicationOperationVerifier, ApplicationOperationVerifierReport,
+};
+use hyperlane_warp_route::TokenMessage;
+
+const WARP_ROUTE_MARKER: &str = "/";
```
[the remark this answers is not in the document]
Comment (yjamin) [reply rust/main/chains/hyperlane-aleo/src/application.rs]: This is not true.

**Review by ameten — APPROVED** [review]
